### PR TITLE
IDAM-481 Show The Signed In Users Name

### PR DESCRIPTION
**Pull request by imranazad**

### CODE REVIEW

File: NICE.Identity.Management/ClientApp/src/components/App/App.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 
 import { BrowserRouter as Router, Route, Redirect } from "react-router-dom";
-import { Header, Footer, IdamProviderProps } from "@nice-digital/global-nav";
+import { Header, Footer, IdamProviderProps, Link } from "@nice-digital/global-nav";
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/components/App/App.tsx:4]: Don't think 'Link' is being used.

File: NICE.Identity.Management/ClientApp/src/data/endpoints.ts [hunk 2]
```diff
@@ -2,6 +2,8 @@ const API_BASE_URL = process.env.REACT_APP_API_BASE_URL
 	? process.env.REACT_APP_API_BASE_URL
 	: "#{REACT_APP_API_BASE_URL}";
 
+const APP_BASE_URL = API_BASE_URL.replace("/api", '');
+
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/data/endpoints.ts:6]: What does this do exactly?

File: NICE.Identity.Management/ClientApp/src/components/App/App.tsx [hunk 3]
```diff
@@ -18,21 +18,61 @@ import { SelectRoles } from "./../../views/SelectRoles/SelectRoles";
...
-	render(): JSX.Element {
+	state = {
+		isLoading: false,
+		auth: {} as IdamProviderProps
+	}
+	async componentDidMount(): Promise<void> {
+		const fetchOptions = {
+			method: "GET",
+			headers: { "Content-Type": "application/json" }
+		};
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/components/App/App.tsx:39]:
```suggestion
               this.setState({ isLoading: true });
               
		const fetchOptions = {
			method: "GET",
			headers: { "Content-Type": "application/json" }
		};
```

Move the isLoading: true from below up here.

File: NICE.Identity.Management/ClientApp/src/components/App/App.tsx [hunk 4]
```diff
@@ -18,21 +18,61 @@ import { SelectRoles } from "./../../views/SelectRoles/SelectRoles";
...
+			auth: auth
+		});
+
+		if (isDataError(myAccountDetails)) {
+			this.setState({ error: myAccountDetails });
+		}
+
+		this.setState({
+			isLoading: false
+		});
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/components/App/App.tsx:66]:
```suggestion
		this.setState({
			auth: auth,
			isLoading: false
		});
```

Move the setState auth from above down here.

File: NICE.Identity.Management/ClientApp/src/components/App/App.tsx [hunk 5]
```diff
@@ -18,21 +18,61 @@ import { SelectRoles } from "./../../views/SelectRoles/SelectRoles";
...
 			],
-			displayName: "John",
+			displayName:  myAccountDetails.displayName,
 			provider: "idam",
 		};
 
+		this.setState({
+			isLoading: true,
+			auth: auth
+		});
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/components/App/App.tsx:58]:
```suggestion
```

Remove this and put bits above and below.

File: NICE.Identity.Management/ClientApp/src/components/App/App.tsx [hunk 6]
```diff
@@ -18,21 +18,61 @@ import { SelectRoles } from "./../../views/SelectRoles/SelectRoles";
...
+
+		if (isDataError(myAccountDetails)) {
+			this.setState({ error: myAccountDetails });
+		}
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/components/App/App.tsx:62]:
```suggestion
```

Move this above.

File: NICE.Identity.Management/ClientApp/src/components/App/App.tsx [hunk 7]
```diff
@@ -18,21 +18,61 @@ import { SelectRoles } from "./../../views/SelectRoles/SelectRoles";
...
+	async componentDidMount(): Promise<void> {
+		const fetchOptions = {
+			method: "GET",
+			headers: { "Content-Type": "application/json" }
+		};
+		const myAccountDetails = await fetchData(
+			Endpoints.identityManagementUser,
+			fetchOptions,
+		) as MyAccountDetails;
+
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/components/App/App.tsx:44]:
```suggestion
		const myAccountDetails = await fetchData(
			Endpoints.identityManagementUser,
			fetchOptions,
		) as MyAccountDetails;
		
		if (isDataError(myAccountDetails)) {
			this.setState({ error: myAccountDetails });
		}
```

Move here from below.

**Review by johnholland-nice — CHANGES_REQUESTED** [review]
Just a few changes. I think the indentation is off for the loading header statement too. 

Give me a shout if you want to discuss. :)

File: NICE.Identity.Management/ClientApp/src/data/endpoints.ts (hunk 2 above)
Comment (imranazad) [reply NICE.Identity.Management/ClientApp/src/data/endpoints.ts:6]: It accesses the account controller to get the status endpoint which essentially returns JSON data containing the username.

Comment (imranazad) [reply NICE.Identity.Management/ClientApp/src/data/endpoints.ts:6]: I've added it as a comment for future reference.

**Review by suegarner — DISMISSED** [review]
Approved AccountController.cs

**Review by johnholland-nice — DISMISSED** [review]

**Review by johnholland-nice — DISMISSED** [review]

**Review by johnholland-nice — APPROVED** [review]

**Pull request merged**